model/model.py

```diff
@@ -0,0 +1,49 @@
+"""Define the main MeshGraphNet model"""
+
+import numpy as np
+import torch
+
+from meshnets.datatypes.graph import Graph
+from meshnets.model.encoder import GraphEncoder
+from meshnets.model.decoder import GraphDecoder
+from meshnets.model.processor import GraphProcessor
+
+
+class MeshGraphNet(torch.nn.Module):
+    """MeshGraphNet model.
+    
+    It is composed of a GraphEncoder, a GraphProcessor and a GraphDecoder.
+    It takes as input Graph objects, encodes their node and mesh features
+    to a latent representation, apply message passing steps
+    on the latent Graph objects, and decodes the output features
+    at each node."""
+
+    def __init__(self, node_feats_size, mesh_feats_size, output_size,
+                 latent_size, num_mlp_layers, message_passing_steps):
+        """Initialize the GraphEncoder, GraphProcessor and GraphDecoder
+        of the MeshGraphNet."""
+        super().__init__()
+
+        self.encoder = GraphEncoder(node_feats_size, mesh_feats_size,
+                                    latent_size, num_mlp_layers)
+
+        self.processor = GraphProcessor(latent_size, message_passing_steps)
+
+        self.decoder = GraphDecoder(output_size, latent_size, num_mlp_layers)
+
+    def forward(self, graph_list: np.ndarray[Graph]) -> torch.Tensor:
```

Comment:
Here, you are passing the input is an n-dimensional array of Graphs. Maybe, just it can just be `List[Graph]`, wdyt?

See [here](https://docs.python.org/3.6/library/typing.html).

Reply:
A list probably makes more sense indeed as we don't need array properties here. I will change that!